src/core/thread/mesh_forwarder.cpp

```diff
@@ -1096,6 +1096,18 @@ void MeshForwarder::HandleReceivedFrame(Mac::RxFrame &aFrame)
             LogFrame("Received empty payload frame", aFrame, OT_ERROR_NONE);
         }
 
+#if OPENTHREAD_THREAD_VERSION >= OPENTHREAD_THREAD_VERSION_1_2 && OPENTHREAD_FTD
```

Comment:
The frame might be drop in HandleFragment() or HandleLowpanHC(), how about moving this block into [MeshForwarder::HandleDatagram](https://github.com/openthread/openthread/blob/master/src/core/thread/mesh_forwarder.cpp#L1366)?